src/main/java/games/strategy/internal/persistence/serializable/AaRadarAdvanceProxy.java

```diff
@@ -6,14 +6,20 @@
 
 import javax.annotation.concurrent.Immutable;
 
-import games.strategy.engine.data.GameData;
 import games.strategy.engine.data.IAttachment;
 import games.strategy.persistence.serializable.Proxy;
 import games.strategy.persistence.serializable.ProxyFactory;
 import games.strategy.triplea.delegate.AARadarAdvance;
 
 /**
  * A serializable proxy for the {@link AARadarAdvance} class.
+ *
+ * <p>
+ * This proxy does not serialize the game data owner to avoid a circular reference. Instances of {@link AARadarAdvance}
+ * created from this proxy will always have their game data set to {@code null}. Proxies that may compose instances of
+ * this proxy are required to manually restore the game data in their {@code readResolve()} method via a
+ * context-dependent mechanism.
+ * </p>
```

Comment:
I added a similar comment to all `GameDataComponent` proxies out of caution.  This might be overkill, but I figured it's an important implementation detail.

This would be a no-brainer if we had something like an `AbstractGameDataComponentProxy` superclass for all of these proxies to inherit from.  Then I could simply have the comment in a single location.  However, I've been purposefully avoiding using base classes to reduce duplication because class inheritance introduces very tight coupling in the serialization stream.  I plan on investigating using some type of mixin pattern in the future to reduce duplication in the proxies without introducing the coupling of implementation inheritance.